docs/guide/writing-tests-for-modules.md

```diff
@@ -0,0 +1,163 @@
+- [eslint](https://www.npmjs.com/package/eslint): *ESLint is a tool for identifying and reporting on patterns found in ECMAScript/JavaScript code*
+- [eslint-config-apostrophe](https://www.npmjs.com/package/eslint-config-apostrophe): *An ESLint configuration for ApostropheCMS core and officials modules*
+- [mocha](https://www.npmjs.com/package/mocha): *Simple, flexible, fun JavaScript test framework for Node.js & The Browser*
+```sh
+npm install --save-dev apostrophe eslint eslint-config-apostrophe mocha
+```
+
+You will need to add some new scripts to your `package.json`.
+
+```json
+{
+  "scripts": {
+    "lint": "npm run eslint",
+    "eslint": "eslint .",
+    "test": "npm run lint && mocha"
+  }
+}
+```
+
+When you call `npm test`, it will lint your files according to [eslint-config-apostrophe](https://www.npmjs.com/package/eslint-config-apostrophe) rules and run the tests using [mocha](https://www.npmjs.com/package/mocha).
+
+### Test folder
+
+You will need a `test/package.json` file referencing the repository URL of your module. Please replace `%module-name%` & `%module-repository-url%` with your module informations.
+
+e.g. for the module [@apostrophecms/sitemap](https://github.com/apostrophecms/sitemap) we use 
+
+> `"@apostrophecms/sitemap": "git://github.com/apostrophecms/sitemap.git"`
+> 
+
+```json
+{
+  "/**": "This package.json file is not actually installed.",
+  " * ": "Apostrophe requires that all npm modules to be loaded by moog",
+  " */": "exist in package.json at project level, which for a test is here",
+  "dependencies": {
+    "apostrophe": "^3.26.0",
+    "%module-name%": "%module-repository-url%"
+  }
+}
+```
+
+You don’t want to commit test files artifacts to git, please add the following to your `.gitignore` file
+
+```gitignore
+# Test files
+/test/apos-build
+/test/public
+/test/locales
+```
+
+## Writing tests
+
+We start with a single test file `test/index.js`.
+
+As your tests grows, you can break it down into multiple files based on your liking.
+
+Mocha does not play well with arrow-functions, more info at [https://mochajs.org/#arrow-functions](https://mochajs.org/#arrow-functions)
+
+
+```javascript
+// test/index.js
+const assert = require('assert');
+const t = require('apostrophe/test-lib/util.js');
+
+const getAppConfig = function (options = {}) {
+  return {
+    '@apostrophecms/express': {
+      options: {
+        session: { secret: 'supersecret' }
+      }
+    },
+    '%module-name%': {
+      options: {
+        // pass the required options here
+        ...options
+      }
+    }
+  };
+};
+
+describe('%module-name%', function () {
+  let apos;
+
+  this.timeout(t.timeout);
+
+  after(function () {
+    return t.destroy(apos);
+  });
+
+  it('should be a property of the apos object', async function () {
```

Comment:
What output do you expect from this if the user decides to use the `@apostrophecms/sitemap`? Maybe an example of a passing test and a failing test?